pynitrokey/cli/nk3/piv.py

```diff
@@ -0,0 +1,738 @@
+import datetime
+import sys
+from typing import Optional, Sequence
+
+import click
+import cryptography
+from asn1crypto import x509
```

Comment:
Is there a special reason to use `asn1crypto` over `cryptography.x509`?

Reply:
Because it does not have an API for extracting the [`tbs_certificate_bytes`](https://cryptography.io/en/latest/x509/reference/#cryptography.x509.Certificate.tbs_certificate_bytes) when building a certificate or CSR builder before signing it.

This means that we cannot sign the certificate or the CSRs with the key stored in the device.

Reply:
I too did not like having to add one more dependency for this, but I didn't see another option.

We could contribute that feature to `cryptography`, but that would first require contributing it to the Rust crate is uses underneath (I checked back then and the rust crate did not support doing this either).

Reply:
I see.  I think this is okay, especially for an experimental feature.  I would just consider opening a feature request for `cryptography` (if it does not exist yet) and adding a comment with a link to that issue so that it is clear why we need `asn1crypto` and what would be the requirement for replacing it.

Reply:
https://github.com/pyca/cryptography/issues/11616